### PR TITLE
Added QC module for HiCExplorer

### PR DESCRIPTION
**Pull request by fidelram**
The new QC module, reads the logs file from hicBuildMatrix and puts them together into a html report.

### CODE REVIEW

File: hicexplorer/hicBuildMatrix.py [hunk 1]
```diff
@@ -136,6 +136,12 @@ def parse_arguments(args=None):
                         type=argparse.FileType('w'),
                         required=True)
 
+    parser.add_argument('--outFileNameLog',
+                        help='Output file name for the log file. If not given, a file with the same name as'
+                             'the matrix, ending in .log will be saved',
+                        metavar='FILENAME',
+                        type=argparse.FileType('w'))
+
```
Comment (fidelram) [inline hicexplorer/hicBuildMatrix.py]: @bgruening heads up, now hicBuildMatrix will save a log file. The file name can be given as parameter or if not set, the same name of the output matrix will be used but with .log This change was necessary to make the new hicQC program to work.

**Comment by fidelram** [comment]
The current QC report uses a log file that is saved when creating the matrix. I am happy with the result but the ideal solution is to save the 'log' values as meta data of the matrix, together with the hdf5 file.

The html file that is currently produced is quite simple and there is lot of room for improvement.

**Comment by bgruening** [comment]
Nice!
I will add this as standard output to the galaxy buildmatrix wrapper. What do you think?

**Comment by fidelram** [comment]
What is now a file was before the STDOUT. I think that in galaxy would be nice to have the QC info as output because the user can then see the results. But also, it is needed to save the log file such that later the hicQC can read that file (and many other) to generate the report.

**Comment by bgruening** [comment]
@fidelram do you think hicQC should be a separate tool, or should this be integrated in buildMatrix as an additional output. This might be more useful for the Galaxy user.

**Comment by thomasmanke** [comment]
I would vote for hicQC to be a separate tool.

**Comment by fidelram** [comment]
The current hicQC was planned to read multiple files to generate images like the following:

image

That's why I make it a separate module. However, I think we need to assume that the HiCExplorer users do not know all the tools, so we could make the QC plots a part of the matrix creation such that the results are readily available.

**Comment by fidelram** [comment]
@bgruening now buildHiCmatrix produces QC plots. For this, the argument `--QCfolder`  is now required.

**Comment by fidelram** [comment]
@bgruening do you want to update the galaxy wrappers before I merge?

**Comment by bgruening** [comment]
I can do this earliest this evening. If you can wait I will try otherwise, please merge and I will catch up later.

**Comment by fidelram** [comment]
sure, no problem. My goal is to make a new release and add pip install, but no hurries.

File: setup.py [hunk 2]
```diff
@@ -99,7 +99,6 @@ def checkProgramIsInstalled(self, program, args, where_to_download,
     author='Fidel Ramirez, Vivek Bhardwaj, Björn Grüning',
     author_email='deeptools@googlegroups.com',
     packages=['hicexplorer'],
-    package_data={'': ['config/deeptools.cfg']},
     scripts=['bin/findRestSite', 'bin/hicBuildMatrix', 'bin/hicCorrectMatrix',
```
Comment (fidelram) [inline setup.py]: What was that line doing there! typical copy & paste error.

Comment (bgruening) [inline setup.py]: :) I hope it is now fixed. The version number looks wired and is not supported with newer pip versions. We need to change this before pushing to pypi.

**Comment by bgruening** [comment]
ping @fidelram

**Comment by fidelram** [comment]
@bgruening shall I merge?

**Comment by bgruening** [comment]
Jupp, it's green :)

**Pull request merged**